kratos/processes/levelset_convection_process.h

```diff
@@ -151,6 +150,38 @@ class LevelSetConvectionProcess
     ///@name Operations
     ///@{
 
+    void ExecuteInitialize() override
```

Comment:
I'd avoid adding the `ExecuteInitialize`  to make clear that the use of the process is always through the `Execute`. Creating a protected method, namely `SetUpLevelSetConvection` process, should be enough.

Reply:
This has not been amended yet.

Reply:
I can do that if you prefer it so, but to me it looks slightly better to have a ExecuteInitialize method that is actually called from outside. I see this call 
https://github.com/KratosMultiphysics/Kratos/blob/95ee4fd0995afbc40a39ea3f56168a528dfd6698/kratos/processes/levelset_convection_process.h#L196
more like a "backward_compatibility" solution. The main reason is that I think in general, its better to explicitly call methods rather than leave the responsibility to other methods that in principle have nothing to do. (Execute should in theory just advect the level set function).

Having said so, I can change this to avoid further fighting : P

Reply:
> Having said so, I can change this to avoid further fighting : P

![image](https://user-images.githubusercontent.com/25484702/128691438-7dfa6e9d-991b-43ff-9607-cc20734d8457.png)

Reply:
No. `Execute` is intended (and assumed) to call all the process methods (or a customary implementation as we do in here). What can easily lead the user to misunderstandings is mixing the "sequential" (ExecuteInitialize, ExecuteInitializeSolutionStep, ...) with the "complete" Execute call.

Reply:
fair enough